tests/test_secrets_manager.py

```diff
@@ -0,0 +1,317 @@
+        ("blahCONVERTTOSEMICOLONblah", "blah;blah", "semi-colon"),
+        ("blahCONVERTTOCOLONblah", "blah:blah", "colon"),
+    ],
+)
+def test__get_secret_converts_special_annotations(
+    test_value, expected_value, test_description
+):
+    v = Vault()
+    v.set_internal_secret("mykey", test_value)
+    actual = v.get_secret("mykey")
+    assert actual == expected_value
+
+
+def test_set_boto_session():
+    v = Vault()
+    expected_session = "blah"
+    v.set_boto_session(expected_session)
+    actual_session = v.get_boto_session()
+    assert actual_session == expected_session
+
+
+def test_search_aws_param_store_for_secret__raises_error_if_no_boto_session():
+    vault = Vault()
+    with pytest.raises(NoConnectedBotoSessionError, match="'blah'"):
+        vault.search_aws_param_store_for_secret("blah")
+
+
+def test_search_aws_param_store_for_secret__raises_error_if_no_param_store_path_supplied():
+    vault = Vault()
+    vault.set_boto_session("blah session")
+    with pytest.raises(NoAwsParameterStorePathError, match="'blah3'"):
+        vault.search_aws_param_store_for_secret("blah3")
+
+
+def test_get_secret__retrieves_value_from_parameter_store(ssm_param):
+    vault, secret_name, expected_secret_value = ssm_param
+
+    actual_secret_value = vault.get_secret(secret_name)
+    assert actual_secret_value == expected_secret_value
+
+
+def test_search_aws_param_store_for_secret__raises_error_if_secret_not_in_parameter_store(
+    vault_for_param_store,
+):
+    vault, _ = vault_for_param_store
+    with pytest.raises(SecretNotFoundInAwsParameterStoreError, match="blah8"):
+        vault.search_aws_param_store_for_secret("test_blah8")
+
+
+def test_search_aws_param_store_for_secret__reraises_permission_error_that_could_occur_during_boto_call(
+    vault_for_param_store,
+):
+    vault, _ = vault_for_param_store
+    with pytest.raises(
+        Exception,
+        match=r"An error occurred \(AccessDeniedException\) when calling the GetParameter operation",
+    ):
+        vault.search_aws_param_store_for_secret("blah22091")
+
+
+def test_get_secret__raises_error_if_secret_not_in_parameter_store(
+    vault_for_param_store,
+):
+    vault, _ = vault_for_param_store
+    with pytest.raises(SecretNotFoundInVaultError, match="test_blah891"):
+        vault.get_secret("blah891")
+
+
+def test_get_secret__works_with_int_secret(vault_for_param_store):
+    vault, _ = vault_for_param_store
+    int_secret = vault.get_secret("int_secret")
+    assert isinstance(int_secret, int) is True
+
+
+def test_get_secret__works_with_float_secret(vault_for_param_store):
+    vault, _ = vault_for_param_store
+    int_secret = vault.get_secret("float_secret")
+    assert isinstance(int_secret, float) is True
+
+
+def test_Vault__raises_warning_when_getting_and_setting_secret_with_kebab_case_name():
+    v = Vault()
+    with pytest.warns(KebabCaseSecretNameWarning, match="kebab_name"):
+        v.set_internal_secret("kebab-name", "dummy_val")
```

Comment:
break this into two separate tests. there's no extensive setup that would be duplicated, so it's always best to just test one thing. having to put 'and' in the name of a test is a 'code smell' to confirm if your test might be unduly testing too many concepts at once